src/Microsoft.ML.AutoML/ColumnInference/ColumnPurpose.cs

```diff
@@ -13,6 +13,7 @@ internal enum ColumnPurpose
         TextFeature = 4,
         Weight = 5,
         ImagePath = 6,
-        SamplingKey = 7
+        SamplingKey = 7,
+        LabelFeature,   // CategoricalFeature that requires ValueToKey converter, better naming?
```

Comment:
Should follow the same pattern as above and explicitly give this a value.

Reply:
Agreed. 

For other places in the ML.NET codebase, this helps with backwards compatibility. The numeric value gets serialized in the model, then read by the newer code, which may have additional values inserted or old values removed from the enum. Without the numeric form, the enum will get deserialized to the wrong value.

Reply:
#resolved.